var/spack/repos/builtin/packages/coin3d/package.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Coin3d(AutotoolsPackage):
+    """Coin is an OpenGL-based, 3D graphics library that has its roots in the
+       Open Inventor 2.1 API, which Coin still is compatible with."""
+
+    homepage = "https://github.com/coin3d/coin"
+    url      = "https://github.com/coin3d/coin/archive/Coin-4.0.0.tar.gz"
+
+    version('4.0.0', sha256='b00d2a8e9d962397cf9bf0d9baa81bcecfbd16eef675a98c792f5cf49eb6e805')
+    version('3.1.0', sha256='70dd5ef39406e1d9e05eeadd54a5b51884a143e127530876a97744ca54173dc3')
+    version('3.0.0', sha256='d5c2eb0ecaa5c83d93daf0e9e275e58a6a8dfadc74c873d51b0c939011f81bfa')
+    version('2.0.0', sha256='6d26435aa962d085b7accd306a0b478069a7de1bc5ca24e22344971852dd097c')
+
+    depends_on('boost@1.45.0:', type='build')
+    depends_on('doxygen', when='+html')
+    depends_on('perl', when='+html')
```

Comment:
Do we need a dependency on OpenGL?

Reply:
There are a lot of "hidden" dependencies in this package: X11, OpenGL, etc. If I'm reading the documentation correctly, they are linked and loaded at run-time, rather than at build-time. I'm not sure how to do this correctly.

Reply:
Hmm, I would still add those hidden dependencies as type=link/run then. Is there any way to tell it where to search at run-time, or does it rely on LD_LIBRARY_PATH? It's a shame we can't RPATH things.

Reply:
It uses `LD_LIBRARY_PATH`.

Reply:
https://gist.github.com/iarspider/6bd019ee01f4c1de527236e8896ed9d5